### PR TITLE
[REVIEW] Explicit mutable_column_view to column_view conversion

### LINKED ISSUE #4112: [BUG] Make mutable_column_view conversion to column_view explicit
**Describe the bug**
Could we make the `mutable_column_view` conversion to `column_view` explicit ?
from:
https://github.com/rapidsai/cudf/blob/51f7f4aa619c43eed829208c1a0ee70b86b13479/cpp/include/cudf/column/column_view.hpp#L499
to:
`explicit operator column_view() const;`

**Steps/Code to reproduce bug**
1. Call
```
auto column = cudf::make_numeric_column(...);
cudf::experimental::fill(column->mutable_view(), 0, column->size(), scalar);
````

**Expected behavior**
I intended to call 
https://github.com/rapidsai/cudf/blob/51f7f4aa619c43eed829208c1a0ee70b86b13479/cpp/include/cudf/filling.hpp#L53-L54
and expected a compile error by compiling the above code but due to the implicit casting the above code compiles just fine and ends up calling https://github.com/rapidsai/cudf/blob/51f7f4aa619c43eed829208c1a0ee70b86b13479/cpp/include/cudf/filling.hpp#L79-L82

### PR DESCRIPTION
**Pull request by jeanp413**
Fixes #4112 

Make the mutable_column_view to column_view conversion explicit to avoid unexpected overload resolution, e.g.:
```
auto column = cudf::make_numeric_column(...);
cudf::experimental::fill(column->mutable_view(), 0, column->size(), scalar);
```

### CODE REVIEW

File: cpp/src/strings/strings_scalar_factories.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ std::unique_ptr<scalar> make_string_scalar(
     cudaStream_t stream,
     rmm::mr::device_memory_resource* mr)
 {
-    auto s = new string_scalar(string, false, stream, mr);
+    auto s = new string_scalar(string, true, stream, mr);
```
Comment (rgsl888prabhu) [inline cpp/src/strings/strings_scalar_factories.cpp]: Why this change ?

Comment (jeanp413) [reply cpp/src/strings/strings_scalar_factories.cpp]: My bad, already reverted that change

**Comment by rgsl888prabhu** [comment]
@jeanp413 It seems there is still some changes left to be made, the CI is failing.

**Comment by harrism** [comment]
See discussion on #4112 : we decided it would be to rename in-place `fill` to `fill_in_place` to prevent unexpected overload resolution.

**Comment by jeanp413** [comment]
I created a new PR #4130 that renames the in-place `fill` function.
